File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] charCount = new int[128];
        int result = 0;
        boolean isOdd = false;

        for (char c : s.toCharArray()) {
            charCount[c]++;
        }

        for (int count : charCount) {
            if (count % 2 == 0) result += count;
            else {
                result += count - 1;
                isOdd = true;
            }
        }

        if (isOdd) result += 1;

        return result;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.kt
class Solution {
    fun longestPalindrome(s: String): Int {
        val charCount = IntArray(128)
        s.forEach { charCount[it.toInt()]++ }
        val result = charCount.sumOf { it / 2 * 2 }
        return if (result < s.length) result + 1 else result
    }
}